### ISSUE TITLE
refactor(v2): rewrite pyrcel.thermo in JAX with numerical stability improvements

### ISSUE THREAD

**Issue by darothen** [darothen/pyrcel#35]
## Context

`pyrcel/thermo_jax.py` (to be renamed `pyrcel/thermo.py` in issue #33) contains faithful JAX translations of the NumPy thermodynamics functions. This issue improves numerical stability and cleans up the implementation, with the `Seq` function receiving the most significant treatment.

### Numerical stability

- [ ] Refactor `Seq` to avoid catastrophic cancellation in `r³ - r_dry³` when `r ≈ r_dry`:

  Original (cancellation-prone):
  ```python
  B_full = (r**3 - r_dry**3) / (r**3 - r_dry**3 * (1.0 - kappa))
  ```

  Refactored (difference-of-cubes factorization):
  ```python
  # r³ - r_dry³ = (r - r_dry)(r² + r·r_dry + r_dry²)
  # This avoids subtracting two nearly-equal large numbers when r ≈ r_dry.
  dr = r - r_dry
  sum_sq = r**2 + r * r_dry + r_dry**2
  num = dr * sum_sq
  den = dr * sum_sq + kappa * r_dry**3
  B_full = num / den
  ```

  **Include a comment in the code explaining the original form and the reason for the reformulation.**

- [ ] Apply the same principle to `Seq_approx` and `equilibrate_jax.py` wherever `r³ - r_dry³` appears

### General cleanup

- [ ] Ensure all functions have complete numpydoc docstrings (Parameters, Returns, See Also)
- [ ] Remove any stale references to `thermo_jax` in comments/docstrings
- [ ] Audit all `jnp.where` uses for correctness under JAX tracing

## Testing

- [ ] Existing `tests/test_thermo_jax.py` equivalence tests must continue to pass
- [ ] Add a test specifically for the `Seq` numerical refactor: verify that the refactored form matches the original to within 1 ULP over a range of `r / r_dry` ratios from 1.0001 to 100

## Notes

- float64 is mandatory (`jax.config.update('jax_enable_x64', True)` at module import) — document why in the module docstring with explicit reference to the `r³ - r_dry³` cancellation
- The numerical refactor also improves float32 accuracy, which is relevant to future GPU support (issue for GPU TBD)
